designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/QrCodeBlock.kt

```diff
@@ -0,0 +1,64 @@
+package org.hisp.dhis.mobile.ui.designsystem.component
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredWidthIn
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import org.hisp.dhis.mobile.ui.designsystem.component.internal.qr.rememberQrCodeGenerator
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+
+/**
+ * Container to render QR code image and text for given
+ * data
+ *
+ * @param data: Data to render QR code and text for
+ */
+@Composable
+fun QrCodeBlock(
+    data: String,
+    modifier: Modifier = Modifier,
+    qrCodeSize: Dp = 240.dp,
+) {
+    if (data.isNotBlank()) {
```

Comment:
We might want to add an error view here, but one is not present in the design

Reply:
this component will only be used to show the QR, the error will be managed from QR code input that is shown in figma, other than that maybe you could add that hardcoded size into the internalSizeValues object and use it from there?